server.js

```diff
@@ -1,38 +1,84 @@
-// to get started!
-// 
-// import goldenGlobesData from './data/golden-globes.json'
-// import avocadoSalesData from './data/avocado-sales.json'
-// import booksData from './data/books.json'
-// import netflixData from './data/netflix-titles.json'
-// import topMusicData from './data/top-music.json'
-
-const mongoUrl = process.env.MONGO_URL || "mongodb://localhost/project-mongo"
-mongoose.connect(mongoUrl, { useNewUrlParser: true, useUnifiedTopology: true })
-mongoose.Promise = Promise
+import express from 'express';
+import bodyParser from 'body-parser';
+import cors from 'cors';
+import mongoose from 'mongoose';
+ 
+import goldenGlobesData from './data/golden-globes.json';
+
+const mongoUrl = process.env.MONGO_URL || "mongodb://localhost/project-mongo";
+mongoose.connect(mongoUrl, { useNewUrlParser: true, useUnifiedTopology: true });
+mongoose.Promise = Promise;
 
 // Defines the port the app will run on. Defaults to 8080, but can be 
 // overridden when starting the server. For example:
 //
 //   PORT=9000 npm start
-const port = process.env.PORT || 8080
-const app = express()
+const port = process.env.PORT || 8080;
+const app = express();
+
+//Middlewares to enable cors and json body parsing
+app.use(cors());
+app.use(bodyParser.json());
+
+const Movie = new mongoose.model('Movie', {
+  year_film: Number,
+  year_award: Number,
+  ceremony: Number,
+  category: String,
+  nominee: String,
+  film: String,
+  win: Boolean
+});
 
-// Add middlewares to enable cors and json body parsing
-app.use(cors())
-app.use(bodyParser.json())
+//To avoid that the data duplicates everytime you seed the database, we have to do deleteMany before you seed the database.
+//Here we say if we reset the database by doing RESET_DB=true npm run dev, then delete the data and seed the database.
+if (process.env.RESET_DB) {
+  const seedDatabase = async () => {
+    await Movie.deleteMany();
 
-// Start defining your routes here
+//Here we take all the data from the whole array and for each item we create a Movie model
+    goldenGlobesData.forEach(item => {
+      const newMovie = new Movie(item);
+      newMovie.save();
+    });
+  }
+  seedDatabase();
+}
+
+// Routes
 app.get('/', (req, res) => {
-  res.send('Hello world')
-})
+  res.send('Welcome to Golden Globes API where you can find information about all the movies nominated')
+});
+
+//This returns all movie nominations. Then we have a universal query param to filter based on any property.
+//For example you can search http://localhost:8080/movies?win=true&year_film=2009 and get all the movies that won for films released in 2009
+app.get('/movies', async (req, res) => {
+  const allMovies = await Movie.find(req.query);
 
-// Start the server
+  if (allMovies.length > 0) {
+    res.json(allMovies);
+  } else {
+    res.status(400).json({ error: 'Invalid entry' });
+  }
+});
+
+//This returns one single movie based on the name of the nomineed movie
+app.get('/movies/nominee/:nominee', async (req, res) => {
```

Comment:
`nominee` is not a good key for extracting single document, unfortunately. When you use it like this you returning a bad set of data because you cutting off a big portion of it. For example "Avatar" has a lot of nominations, but you are showing only the first one. 

You need to use field which is always unique for every document, usually, it is `id` field.

Reply:
Replaced it with id.

Reply:
@KarolCislowski  I see that this project is still not approved. Are you missing something?